tests: net.ptp.clock: frdm_k64f:  build failure

**Describe the bug**

net.ptp.clock build failure

Please also mention any information which could help others to understand
the problem you're facing:
 - What target platform are you using?
 - frdm_k64f
 - What have you tried to diagnose or workaround this issue?
 -  no
 - Is this a regression? If yes, have you been able to "git bisect" it to a
   specific commit?
 - git bisect find this is introduced by

```

commit 537d5c310c3e9d22f1df9a6d0bd1e250c84216fb
Author: Declan Snyder <declan.snyder@nxp.com>
Date:   Mon Mar 18 18:42:55 2024 -0500

    dts: nxp: Convert ENET DT default to new binding.
    
    Convert all of the NXP SOCs with ENET to use the new
    binding scheme, which is used by the new driver.
    
```

**To Reproduce**
Steps to reproduce the behavior:
west build -b frdm_k64f tests/net/ptp/clock

**Expected behavior**
build pass

**Impact**
PTP function

**Logs and console output**

```
/home/shared/disk/zephyr_project/zephyr_test/zephyr/drivers/ptp_clock/ptp_clock_nxp_enet.c:19:10: fatal error: fsl_enet.h: No such file or directory
   19 | #include "fsl_enet.h"
      |          ^~~~~~~~~~~~
compilation terminated.
[118/190] Building C object zephyr/subsys/net/ip/CMakeFiles/subsys__net__ip.dir/ipv6_nbr.c.obj
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /home/ubuntu/.local/bin/cmake --build /home/shared/disk/zephyr_project/zephyr_test/zephyr/build
```

**Environment (please complete the following information):**
 - OS: (e.g. Linux, MacOS, Windows)
 - Toolchain (e.g Zephyr SDK, ...)
 - Commit SHA or Version used:  v3.6.0-2700-ga6400b42feeb

--- a/drivers/ptp_clock/Kconfig.nxp_enet
+++ b/drivers/ptp_clock/Kconfig.nxp_enet
@@ -5,5 +5,6 @@ config PTP_CLOCK_NXP_ENET
 	bool "NXP ENET PTP Clock driver"
 	default y
 	depends on DT_HAS_NXP_ENET_PTP_CLOCK_ENABLED && NET_L2_PTP
+	depends on ETH_NXP_ENET
 	help
 	  Enable NXP ENET PTP clock support.

--- a/drivers/ptp_clock/ptp_clock_nxp_enet.c
+++ b/drivers/ptp_clock/ptp_clock_nxp_enet.c
@@ -16,7 +16,7 @@
 #include <zephyr/drivers/pinctrl.h>
 #include <zephyr/drivers/ethernet/eth_nxp_enet.h>
 
-#include "fsl_enet.h"
+#include <fsl_enet.h>
 
 struct ptp_clock_nxp_enet_config {
 	ENET_Type *base;

--- a/tests/net/ptp/clock/boards/frdm_k64f.overlay
+++ b/tests/net/ptp/clock/boards/frdm_k64f.overlay
@@ -4,9 +4,6 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-&enet {
-
-	ptp {
-		status = "ok";
-	};
+&enet_ptp_clock {
+	status = "okay";
 };

